capture.go

```diff
@@ -4,11 +4,20 @@
 package command
 
 import (
+	"bytes"
 	"context"
 	"errors"
+	"io"
 	"os/exec"
+	"time"
 )
 
+type Pipes struct {
+	Stdin  io.WriteCloser
```

Comment:
These feel backwards, as though they're from a different perspective. To me, StdIn is not a writer, it's a reader, and Stdout / err are writers you would append to for output. From the perspective of an application handling it though, I guess it makes sense. It will be *writing* stdin to another application and will only ever *read* that commands stdout / err

Reply:
These are the types returned by the pipes in `Cmd.Std*Pipe()` and are from its perspective. Note in `doRun` that we actually have a `bytes.Buffer` type, which is a `ReadWriter`. Internally `exec.Cmd` uses similar buffers (in this case a pipe which is a full `File` type) but only exposes one direction from its perspective.

The magic of interfaces.

Reply:
It finally dawned on me where the confusion on this topic is coming from.  It's all about _context_.  Framing.  As Eli has recently commented, (I'm paraphrasing) I/O can be devilishly tricky.  

So, let's remind ourselves what STDIN, STDOUT, and STDERR actually _are_.  They are _standardized_ interfaces to I/O streams, with _standard_ file descriptors (in *NIX) [STDIN always has file descriptor 0, for example], with default I/O devices connected to them [STDIN is connected to the keyboard by default].  When a new process is created, these streams get wired up to that processes I/O facilities by default if nothing else is specified.  They can be overridden through the use of redirects or piping (e.g. pipelines).  So far, so good.

So why are we referring to STDIN as an `io.WriteCloser`, then?  The answer lies in remembering that there are two ends to a stream.  While it absolutely true that _our_ process of interest _reads_ from STDIN, the sending process only ever _writes_ to STDIN.  Similarly, STOUT and STDERR are reversed at the other end of those streams.  Hence the reason everything seems completely backwards from what you might intuit.  These standard streams are a construct of the OS shell, which is providing the I/O interface at one end of the stream/pipe for communicating with our process of interest.  This is the end we are attaching to when we invoke `cmd.Exec()`, etc.